bool button_state
---
string str
